### ISSUE TITLE
Security vulnerabiity: <caps lock> temporarily disables lock screen

### ISSUE THREAD

**Issue by CottonEaster** [linuxmint/cinnamon-screensaver#292]
```
 * Cinnamon-Screensaver version (cinnamon-screensaver --version) 3.8.2
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...): Mint 19
 * Graphics hardware *and* driver used: UHD Graphics 620 (Kabylake GT3); 3.0 Mesa 18.2.1
 * 32 or 64 bit: 64
 ```

**Issue**

Pressing caps lock on the lock screen temporarily disables the lock screen. Or at least it does so when the 'OSD' caps lock indicator (Settings -> Accessibility -> Keyboard -> Use visual indicator) is enabled.

This seem to be a serious security problem.

**Steps to reproduce**

Enable the option mentioned above. Lock the computer. Press caps lock.

**Other information**

OS: Linux Mint 19 Tara x86_64 
Host: 20KHCTO1WW ThinkPad X1 Carbon 6th 
Kernel: 4.19.0-041900rc6-generic 
Uptime: 8 mins 
Packages: 3426 (dpkg), 5 (flatpak) 
Shell: bash 4.4.19 
Resolution: 2560x1440 
DE: Cinnamon 3.8.9 
WM: Mutter (Muffin) 
WM Theme: Mint-Y-Dark (Mint-Y-Dark) 
Theme: Mint-Y-Dark [GTK2/3] 
Icons: Paper [GTK2/3] 
Terminal: xfce4-terminal 
Terminal Font: Consolas 12 
CPU: Intel i7-8550U (8) @ 4.000GHz 
GPU: Intel UHD Graphics 620 
Memory: 2059MiB / 15931MiB 

**EDIT**: the problem is intermittent.

**Comment by MPE2016** [comment]
Hi there CottonEaster,

Right now I cannot reproduce your bug. When I enable the OSD capslock indicator and lock my computer, things seem to work fine (e.g. I get a message that Capslock is on, but that's normal behavior right?). 

Perhaps it is in the way that I lock my computer. I use CTRL+ALT+L. Is that the same as yours?

All the Best,

MPE

**Comment by MPE2016** [comment]
You're welcome CottonEaster,

It would definitely be interesting to see when the problem occurs exactly (idle or closing lid, or perhaps in both cases), so if you can find some time to check it, that would be great!

I did notice that you said the problem was intermittent. Can describe in what way?

**Comment by CottonEaster** [comment]
Actually, I think I've wasted your time. Apologies. For I think what was happening was this: I've got the screensaver set to lock the computer only after 15 seconds; the 'problem' occurred when I pressed <caps lock> within those 15 seconds; but that behaviour is by design - the screensaver should abort, without asking for a password, when a key is pressed before the time-before-lock period has elapsed.
